src/Domain/CoordinateMaps/CylindricalEndcap.hpp

```diff
@@ -0,0 +1,150 @@
+}  // namespace PUP
+/// \endcond
+
+namespace domain::CoordinateMaps {
+
+/*!
+ * \ingroup CoordinateMapsGroup
+ *
+ * \brief Map from 3D unit right cylinder to a volume that connects
+ *  portions of two spherical surfaces.
+ *
+ * \image html CylindricalEndcap.svg "A cylinder maps to the shaded region."
+ *
+ * \details Consider two spheres with centers \f$C_1\f$ and \f$C_2\f$,
+ * and radii \f$R_1\f$ and \f$R_2\f$. Let sphere 1 be intersected by a
+ * plane normal to the \f$z\f$ axis and located at \f$z = z_\mathrm{P}\f$.
+ * Also let there be a projection point \f$P\f$.
+ *
+ * CylindricalEndcap maps a 3D unit right cylinder (with coordinates
+ * \f$(\bar{x},\bar{y},\bar{z})\f$ such that \f$-1\leq\bar{z}\leq 1\f$
+ * and \f$\bar{x}^2+\bar{y}^2 \leq 1\f$) to the shaded area
+ * in the figure above (with coordinates \f$(x,y,z)\f$).  The "bottom"
+ * of the cylinder \f$\bar{z}=-1\f$ is mapped to the portion of sphere
+ * 1 that has \f$z \geq z_\mathrm{P}\f$.  Curves of constant
+ * \f$(\bar{x},\bar{y})\f$ are mapped to portions of lines that pass
+ * through \f$P\f$. Along each of these curves, \f$\bar{z}=-1\f$ is
+ * mapped to a point on sphere 1 and \f$\bar{z}=+1\f$ is mapped to a
+ * point on sphere 2.
+ *
+ * CylindricalEndcap is intended to be composed with Wedge2D maps to
+ * construct a portion of a cylindrical domain for a binary system.
+ *
+ * CylindricalEndcap is described briefly in the Appendix of
+ * Phys. Rev. D 86, 084033 (2012), https://arxiv.org/abs/1206.3015.
+ * CylindricalEndcap is used to construct the blocks labeled 'CA
+ * wedge', 'EA wedge', 'CB wedge', 'EE wedge', and 'EB wedge' in
+ * Figure 20 of that paper.
+ *
+ * CylindricalEndcap is implemented using `FocallyLiftedMap`
+ * and `FocallyLiftedEndcap`; see those classes for details.
+ *
+ * ### Restrictions on map parameters.
+ *
+ * We demand that Sphere 1 is fully contained inside Sphere 2. It is
+ * possible to construct a valid map without this assumption, but the
+ * assumption simplifies the code, and the expected use cases obey
+ * this restriction.
+ *
+ * We also demand that \f$z_\mathrm{P} > C_1^2\f$, that is, the plane
+ * in the above and below figures lies to the right of \f$C_1^2\f$.
+ * This restriction not strictly necessary but is made for simplicity.
+ *
+ * The map is invertible only for some choices of the projection point
+ * \f$P\f$.  Given the above restrictions, the allowed values of
+ * \f$P\f$ are illustrated by the following diagram:
+ *
+ * \image html CylindricalEndcap_Allowed.svg "Allowed region for P." width=75%
+ *
+ * The plane \f$z=z_\mathrm{P}\f$ intersects sphere 1 on a circle. The
+ * cone with apex \f$C_1\f$ that intersects that circle has opening
+ * angle \f$2\theta\f$ as shown in the above figure. Construct another
+ * cone, the "invertibility cone", with apex \f$S\f$ chosen such that
+ * the two cones intersect at right angles on the circle; thus the
+ * opening angle of the invertibility cone is \f$2(\pi-\theta)\f$.  A
```

Comment:
Thanks for making the figures! They are very helpful. However, shouldn't the angle in figure "Allowed region for P" labeled pi - \theta be \pi/2 - theta? Otherwise, the angles in the upper right triangle add up to 3\pi/2.

Similarly, shouldn't the angle of the inevitability cone be \pi - 2\theta?

Reply:
Yes, thanks for catching this!